src/core/stream/buffers_handler.ts

```diff
@@ -708,8 +708,22 @@ function createNativeSourceBuffersForPeriod(
   Object.keys(period.adaptations).forEach(bufferType => {
     if (SourceBufferManager.isNative(bufferType)) {
       const adaptations = period.adaptations[bufferType] || [];
+      const mainAdaptation = adaptations.find((adaptation) => {
+        if (
+          adaptation.role &&
+          adaptation.role.value === "main" &&
+          adaptation.role.schemeIdUri === "urn:mpeg:dash:role:2011"
+        ) {
```

Comment:
same, maybe we should merge the main adaptations already in 
``src/parsers/manifest/dash``